<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>HiContacts: R interface to cool files â€¢ HiContacts</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="HiContacts: R interface to cool files">
<meta property="og:description" content="HiContacts: R interface to cool files.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">HiContacts</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.9.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/js2264/HiContacts/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div class="section level1">
<div class="page-header"><h1 id="hicontacts">HiContacts<a class="anchor" aria-label="anchor" href="#hicontacts"></a>
</h1></div>
<!-- badges: start -->
<p>HiContacts provides tools to import <code>(m)cool</code> matrices in R and work with them there.</p>
<p>Rather than creating redundant classes, it relies on pre-existing Bioconductor objects, namely <code>InteractionSet</code>, <code>GenomicInterations</code> and <code>ContactMatrix</code> (<code>Lun, Perry &amp; Ing-Simmons, F1000Research 2016</code>).</p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">'js2264/HiContacts'</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="import-a-cool-file-as-genomicinterations">Import a .cool file as GenomicInterations<a class="anchor" aria-label="anchor" href="#import-a-cool-file-as-genomicinterations"></a>
</h2>
<p><code>cool2gi</code> can be used to import contact counts as <code>GenomicInterations</code>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">file</span> <span class="op">&lt;-</span> <span class="st">'path/to/file.cool'</span></span>
<span><span class="va">range</span> <span class="op">&lt;-</span> <span class="st">'ranges_of_interest'</span> <span class="co"># e.g. range &lt;- 'chrI:2000-8000'</span></span>
<span><span class="va">gis</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/cool2gi.html">cool2gi</a></span><span class="op">(</span><span class="va">file</span>, coords <span class="op">=</span> <span class="va">range</span><span class="op">)</span></span>
<span><span class="va">gis</span></span></code></pre></div>
<p><code>cool2gi</code> works with <code>.mcool</code> files as well: in this case, the user needs to specify the resolution at which count values are recovered.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">file</span> <span class="op">&lt;-</span> <span class="st">'path/to/file.mcool'</span></span>
<span><span class="va">range</span> <span class="op">&lt;-</span> <span class="st">'ranges_of_interest'</span> <span class="co"># e.g. range &lt;- 'chrI:2000-8000' or range &lt;- 'chrI'</span></span>
<span><span class="fu"><a href="reference/lsCoolResolutions.html">lsCoolResolutions</a></span><span class="op">(</span><span class="va">file</span><span class="op">)</span></span>
<span><span class="va">gis</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/cool2gi.html">cool2gi</a></span><span class="op">(</span><span class="va">file</span>, coords <span class="op">=</span> <span class="va">range</span>, res <span class="op">=</span> <span class="fl">1000</span><span class="op">)</span></span>
<span><span class="va">gis</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="plot-matrix-heatmaps">Plot matrix heatmaps<a class="anchor" aria-label="anchor" href="#plot-matrix-heatmaps"></a>
</h2>
<blockquote><p>Diagonal style</p></blockquote>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gis</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/cool2gi.html">cool2gi</a></span><span class="op">(</span><span class="va">file</span>, coords <span class="op">=</span> <span class="va">range</span>, res <span class="op">=</span> <span class="va">res</span><span class="op">)</span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/plotMatrix.html">plotMatrix</a></span><span class="op">(</span><span class="va">gis</span>, limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">3</span>, <span class="op">-</span><span class="fl">1</span><span class="op">)</span>, dpi <span class="op">=</span> <span class="fl">400</span><span class="op">)</span></span></code></pre></div>
<blockquote><p>Plot matrix of correlation</p></blockquote>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/cool2gi.html">cool2gi</a></span><span class="op">(</span><span class="va">file</span>, coords <span class="op">=</span> <span class="va">range</span>, res <span class="op">=</span> <span class="va">res</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>    <span class="fu">autocorrelate</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>    <span class="fu"><a href="reference/plotMatrix.html">plotMatrix</a></span><span class="op">(</span>scale <span class="op">=</span> <span class="st">'linear'</span>, limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<blockquote><p>Plot signal over expected</p></blockquote>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/cool2gi.html">cool2gi</a></span><span class="op">(</span><span class="va">file</span>, coords <span class="op">=</span> <span class="va">range</span>, res <span class="op">=</span> <span class="va">res</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>    <span class="fu"><a href="reference/detrend.html">detrend</a></span><span class="op">(</span>update_scores <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>    <span class="fu"><a href="reference/plotMatrix.html">plotMatrix</a></span><span class="op">(</span>scale <span class="op">=</span> <span class="st">'linear'</span>, cmap <span class="op">=</span> <span class="va">bwr_colors</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="plot-aggregated-matrices">Plot aggregated matrices<a class="anchor" aria-label="anchor" href="#plot-aggregated-matrices"></a>
</h2>
<blockquote><p>At borders (e.g.Â TAD boundaries)</p></blockquote>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">loops</span> <span class="op">&lt;-</span> <span class="fu">rtracklayer</span><span class="fu">::</span><span class="fu">import</span><span class="op">(</span><span class="st">'CH195-loops.bedpe'</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">makeGInteractionsFromGRangesPairs</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co"># - Trim extended borders to only keep full-size GRanges</span></span>
<span><span class="fu">seqlevels</span><span class="op">(</span><span class="va">coords</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">seqlevels</span><span class="op">(</span><span class="fu"><a href="reference/cool2seqinfo.html">cool2seqinfo</a></span><span class="op">(</span><span class="va">file</span>, res <span class="op">=</span> <span class="va">res</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">seqinfo</span><span class="op">(</span><span class="va">coords</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/cool2seqinfo.html">cool2seqinfo</a></span><span class="op">(</span><span class="va">file</span>, res <span class="op">=</span> <span class="va">res</span><span class="op">)</span></span>
<span><span class="va">coords</span> <span class="op">&lt;-</span> <span class="fu">trim</span><span class="op">(</span><span class="va">coords</span><span class="op">)</span></span>
<span><span class="va">coords</span> <span class="op">&lt;-</span> <span class="va">coords</span><span class="op">[</span><span class="fu">width</span><span class="op">(</span><span class="va">coords</span><span class="op">)</span> <span class="op">==</span> <span class="fl">2000000</span><span class="op">]</span></span>
<span><span class="co"># - Plot aggregated signal over borders</span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu">plotAggregatedMatrix</span><span class="op">(</span></span>
<span>    <span class="va">file</span>, </span>
<span>    <span class="va">coords</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span>, </span>
<span>    res <span class="op">=</span> <span class="va">res</span>, </span>
<span>    BPPARAM <span class="op">=</span> <span class="fu">BiocParallel</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/MulticoreParam-class.html" class="external-link">MulticoreParam</a></span><span class="op">(</span>workers <span class="op">=</span> <span class="fl">2</span>, tasks <span class="op">=</span> <span class="fl">200</span>, progressbar <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggsave.html" class="external-link">ggsave</a></span><span class="op">(</span><span class="st">'plot.png'</span>, width <span class="op">=</span> <span class="fl">10</span>, height <span class="op">=</span> <span class="fl">10</span>, dpi <span class="op">=</span> <span class="fl">300</span><span class="op">)</span></span></code></pre></div>
<blockquote><p>On pairs of coordinates (e.g.Â structural loops)</p></blockquote>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">file</span> <span class="op">&lt;-</span> <span class="st">'path/to/file.mcool'</span></span>
<span><span class="co"># - Get loop coordinates from `bedpe` file as an `InteractionSet` object</span></span>
<span><span class="va">loops</span> <span class="op">&lt;-</span> <span class="fu">rtracklayer</span><span class="fu">::</span><span class="fu">import</span><span class="op">(</span><span class="st">'path/to/loops.bedpe'</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="co">## interactions as `Pairs` object</span></span>
<span>    <span class="fu">GenomicRanges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/intra-range-methods.html" class="external-link">resize</a></span><span class="op">(</span>width <span class="op">=</span> <span class="fl">30000</span>, fix <span class="op">=</span> <span class="st">'center'</span><span class="op">)</span></span>
<span><span class="co"># - Only retain loops for which extended anchors are both within the genome</span></span>
<span><span class="va">loops</span> <span class="op">&lt;-</span> <span class="va">loops</span><span class="op">[</span><span class="fu">InteractionSet</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/InteractionSet/man/interaction-accessors.html" class="external-link">anchors</a></span><span class="op">(</span><span class="va">loops</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op">%within%</span> <span class="fu">GRanges</span><span class="op">(</span><span class="fu"><a href="reference/cool2seqinfo.html">cool2seqinfo</a></span><span class="op">(</span><span class="va">file</span><span class="op">)</span><span class="op">)</span> <span class="op">&amp;</span> <span class="fu">InteractionSet</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/InteractionSet/man/interaction-accessors.html" class="external-link">anchors</a></span><span class="op">(</span><span class="va">loops</span><span class="op">)</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span> <span class="op">%within%</span> <span class="fu">GRanges</span><span class="op">(</span><span class="fu"><a href="reference/cool2seqinfo.html">cool2seqinfo</a></span><span class="op">(</span><span class="va">file</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="co"># - Plot aggregated signal over sets of loop coordinates</span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu">plotAggregatedMatrix</span><span class="op">(</span><span class="va">file</span>, coords <span class="op">=</span> <span class="va">loops</span>, symmetrical <span class="op">=</span> <span class="cn">FALSE</span>, BPPARAM <span class="op">=</span> <span class="fu">BiocParallel</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/MulticoreParam-class.html" class="external-link">MulticoreParam</a></span><span class="op">(</span>workers <span class="op">=</span> <span class="fl">16</span>, tasks <span class="op">=</span> <span class="fl">200</span>, progressbar <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggsave.html" class="external-link">ggsave</a></span><span class="op">(</span><span class="st">'plot.png'</span>, width <span class="op">=</span> <span class="fl">10</span>, height <span class="op">=</span> <span class="fl">10</span>, dpi <span class="op">=</span> <span class="fl">300</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="display-features-on-matrices">Display features on matrices<a class="anchor" aria-label="anchor" href="#display-features-on-matrices"></a>
</h2>
<blockquote><p>Plot TADs on the matrix</p></blockquote>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gis</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/cool2gi.html">cool2gi</a></span><span class="op">(</span><span class="va">file</span>, coords <span class="op">=</span> <span class="va">range</span>, res <span class="op">=</span> <span class="va">res</span><span class="op">)</span></span>
<span><span class="va">tads</span> <span class="op">&lt;-</span> <span class="fu">rtracklayer</span><span class="fu">::</span><span class="fu">import</span><span class="op">(</span><span class="st">'path/to/tads.bed'</span><span class="op">)</span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/plotMatrix.html">plotMatrix</a></span><span class="op">(</span><span class="va">gis</span>, dpi <span class="op">=</span> <span class="fl">500</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>    <span class="fu">addTads</span><span class="op">(</span><span class="va">tads</span>, coords <span class="op">=</span> <span class="va">range</span><span class="op">)</span></span></code></pre></div>
<blockquote><p>Plot loops on the matrix</p></blockquote>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gis</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/cool2gi.html">cool2gi</a></span><span class="op">(</span><span class="va">file</span>, coords <span class="op">=</span> <span class="va">range</span>, res <span class="op">=</span> <span class="va">res</span><span class="op">)</span></span>
<span><span class="va">loops</span> <span class="op">&lt;-</span> <span class="fu">rtracklayer</span><span class="fu">::</span><span class="fu">import</span><span class="op">(</span><span class="st">'path/to/loops.bedpe'</span><span class="op">)</span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/plotMatrix.html">plotMatrix</a></span><span class="op">(</span><span class="va">gis</span>, dpi <span class="op">=</span> <span class="fl">500</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>    <span class="fu">addTads</span><span class="op">(</span><span class="va">tads</span>, coords <span class="op">=</span> <span class="va">range</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>    <span class="fu">addLoops</span><span class="op">(</span><span class="va">loops</span>, coords <span class="op">=</span> <span class="va">range</span><span class="op">)</span></span></code></pre></div>
<blockquote><p>Plot genebodies and specific coordinates underneath a matrix</p></blockquote>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## -- Generate heatmap</span></span>
<span><span class="va">file</span> <span class="op">&lt;-</span> <span class="st">'path/to/file.mcool'</span></span>
<span><span class="va">range</span> <span class="op">&lt;-</span> <span class="st">'chr13:110000000-115000000'</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fl">20000</span></span>
<span><span class="va">gis</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/cool2gi.html">cool2gi</a></span><span class="op">(</span><span class="va">file</span>, coords <span class="op">=</span> <span class="va">range</span>, res <span class="op">=</span> <span class="va">res</span><span class="op">)</span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/plotMatrix.html">plotMatrix</a></span><span class="op">(</span><span class="va">gis</span>, limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">3</span>, <span class="op">-</span><span class="fl">1</span><span class="op">)</span>, dpi <span class="op">=</span> <span class="fl">500</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## -- Import gene annotations</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">plyranges</span><span class="op">)</span></span>
<span><span class="va">mm10_genes</span> <span class="op">&lt;-</span> <span class="fu">AnnotationHub</span><span class="fu">::</span><span class="fu">query</span><span class="op">(</span><span class="fu">AnnotationHub</span><span class="fu">::</span><span class="fu">AnnotationHub</span><span class="op">(</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'Mus_musculus.GRCm39.104.gtf'</span><span class="op">)</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op">%&gt;%</span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/r/stats/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">gene_biotype</span> <span class="op">==</span> <span class="st">'protein_coding'</span>, <span class="va">source</span> <span class="op">==</span> <span class="st">'ensembl_havana'</span>, <span class="va">type</span> <span class="op">==</span> <span class="st">'gene'</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>    <span class="fu">select</span><span class="op">(</span><span class="va">gene_id</span>, <span class="va">gene_name</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>    <span class="fu">mutate</span><span class="op">(</span>ID <span class="op">=</span> <span class="va">gene_name</span><span class="op">)</span></span>
<span><span class="fu">seqlevelsStyle</span><span class="op">(</span><span class="va">mm10_genes</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">'UCSC'</span></span>
<span></span>
<span><span class="co">## -- Import CTCF binding sites</span></span>
<span><span class="co"># mm10_CTCF &lt;- rtracklayer::import('http://hgdownload.soe.ucsc.edu/gbdb/mm10/encode3/ccre/encodeCcreCombined.bb') %&gt;% </span></span>
<span><span class="va">mm10_CTCF</span> <span class="op">&lt;-</span> <span class="fu">rtracklayer</span><span class="fu">::</span><span class="fu">import</span><span class="op">(</span><span class="st">'~/encodeCcreCombined.bb'</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/r/stats/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">'CTCF-bound'</span>, <span class="va">ccre</span><span class="op">)</span>, <span class="va">encodeLabel</span> <span class="op">==</span> <span class="st">'CTCF-only'</span><span class="op">)</span> </span>
<span></span>
<span><span class="co">## -- Import profiles</span></span>
<span><span class="va">comps</span> <span class="op">&lt;-</span> <span class="fu">rtracklayer</span><span class="fu">::</span><span class="fu">import</span><span class="op">(</span><span class="st">'~/Projects/20210602_MCCs_HiC-deuts/compartments/AT409_100kb.cis.bw'</span>, as <span class="op">=</span> <span class="st">'Rle'</span><span class="op">)</span></span>
<span><span class="va">insul</span> <span class="op">&lt;-</span> <span class="fu">rtracklayer</span><span class="fu">::</span><span class="fu">import</span><span class="op">(</span><span class="st">'~/Projects/20210602_MCCs_HiC-deuts/tads/AT409_25kb_insulation-scores.bw'</span>, as <span class="op">=</span> <span class="st">'Rle'</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## -- Combine all</span></span>
<span><span class="va">p_withTracks</span> <span class="op">&lt;-</span> <span class="fu">addTracks</span><span class="op">(</span><span class="va">p</span>, <span class="va">range</span>, </span>
<span>    annotations <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>genes <span class="op">=</span> <span class="va">mm10_genes</span>, CTCF <span class="op">=</span> <span class="va">mm10_CTCF</span><span class="op">)</span>, </span>
<span>    profiles <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>eigen <span class="op">=</span> <span class="va">comps</span>, insulation <span class="op">=</span> <span class="va">insul</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggsave.html" class="external-link">ggsave</a></span><span class="op">(</span><span class="st">'plot.png'</span>, plot <span class="op">=</span> <span class="va">p_withTracks</span>, width <span class="op">=</span> <span class="fl">10</span>, height <span class="op">=</span> <span class="fl">10</span>, dpi <span class="op">=</span> <span class="fl">500</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="complex-examples">Complex examples<a class="anchor" aria-label="anchor" href="#complex-examples"></a>
</h2>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tads</span> <span class="op">&lt;-</span> <span class="fu">rtracklayer</span><span class="fu">::</span><span class="fu">import</span><span class="op">(</span><span class="st">'path/to/tads.bed'</span><span class="op">)</span></span>
<span><span class="va">loops</span> <span class="op">&lt;-</span> <span class="fu">rtracklayer</span><span class="fu">::</span><span class="fu">import</span><span class="op">(</span><span class="st">'path/to/loops.bedpe'</span><span class="op">)</span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu">cowplot</span><span class="fu">::</span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html" class="external-link">plot_grid</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="reference/plotMatrix.html">plotMatrix</a></span><span class="op">(</span><span class="fu"><a href="reference/cool2gi.html">cool2gi</a></span><span class="op">(</span><span class="va">file</span>, coords <span class="op">=</span> <span class="st">'chr13'</span>, res <span class="op">=</span> <span class="fl">160000</span><span class="op">)</span>, dpi <span class="op">=</span> <span class="fl">500</span>, limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">3</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span>, </span>
<span>    <span class="fu">plotCorrelatedMatrix</span><span class="op">(</span><span class="fu"><a href="reference/cool2gi.html">cool2gi</a></span><span class="op">(</span><span class="va">file</span>, coords <span class="op">=</span> <span class="st">'chr13'</span>, res <span class="op">=</span> <span class="fl">160000</span><span class="op">)</span>, dpi <span class="op">=</span> <span class="fl">500</span><span class="op">)</span>, </span>
<span>    <span class="fu">plotMatrixList</span><span class="op">(</span></span>
<span>        ls <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>            <span class="st">'chr13'</span> <span class="op">=</span> <span class="fu"><a href="reference/cool2gi.html">cool2gi</a></span><span class="op">(</span><span class="va">file</span>, coords <span class="op">=</span> <span class="st">'chr13'</span>, res <span class="op">=</span> <span class="fl">40000</span><span class="op">)</span>, </span>
<span>            <span class="st">'50-115Mb'</span> <span class="op">=</span> <span class="fu"><a href="reference/cool2gi.html">cool2gi</a></span><span class="op">(</span><span class="va">file</span>, coords <span class="op">=</span> <span class="st">'chr13:50000000-115000000'</span>, res <span class="op">=</span> <span class="fl">40000</span><span class="op">)</span>, </span>
<span>            <span class="st">'100-115Mb'</span> <span class="op">=</span> <span class="fu"><a href="reference/cool2gi.html">cool2gi</a></span><span class="op">(</span><span class="va">file</span>, coords <span class="op">=</span> <span class="st">'chr13:100000000-115000000'</span>, res <span class="op">=</span> <span class="fl">40000</span><span class="op">)</span>,</span>
<span>            <span class="st">'111-115Mb'</span> <span class="op">=</span> <span class="fu"><a href="reference/cool2gi.html">cool2gi</a></span><span class="op">(</span><span class="va">file</span>, coords <span class="op">=</span> <span class="st">'chr13:110000000-115000000'</span>, res <span class="op">=</span> <span class="fl">40000</span><span class="op">)</span></span>
<span>        <span class="op">)</span>, </span>
<span>        limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">3</span>, <span class="fl">0</span><span class="op">)</span></span>
<span>    <span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>        <span class="fu">addTads</span><span class="op">(</span><span class="va">tads</span>, coords <span class="op">=</span> <span class="st">'chr13:110000000-115000000'</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>        <span class="fu">addLoops</span><span class="op">(</span><span class="va">loops</span>, coords <span class="op">=</span> <span class="st">'chr13:110000000-115000000'</span><span class="op">)</span>, </span>
<span>    <span class="fu">plotOverExpected</span><span class="op">(</span><span class="fu"><a href="reference/cool2gi.html">cool2gi</a></span><span class="op">(</span><span class="va">file</span>, coords <span class="op">=</span> <span class="st">'chr13:110000000-115000000'</span>, res <span class="op">=</span> <span class="fl">10000</span><span class="op">)</span>, limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="fu">ggsave</span><span class="op">(</span><span class="st">'HiC.png'</span>, width <span class="op">=</span> <span class="fl">12</span>, height <span class="op">=</span> <span class="fl">12</span>, dpi <span class="op">=</span> <span class="fl">500</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/js2264/HiContacts/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/js2264/HiContacts/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing HiContacts</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Jacques Serizay <br><small class="roles"> Author, maintainer </small>  </li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental" class="external-link"><img src="https://img.shields.io/badge/lifecycle-experimental-orange.svg" alt="Lifecycle: experimental"></a></li>
<li><a href="https://github.com/js2264/HiContacts/actions/workflows/check-bioc.yml" class="external-link"><img src="https://github.com/js2264/HiContacts/actions/workflows/check-bioc.yml/badge.svg" alt="R-CMD-check-bioc"></a></li>
<li><a href="https://github.com/js2264/HiContacts/actions" class="external-link"><img src="https://github.com/js2264/HiContacts/workflows/pkgdown/badge.svg" alt="pkgdown"></a></li>
</ul>
</div>

  </div>
</div>


      <footer><div class="copyright">
  <p></p>
<p>Developed by Jacques Serizay.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.5.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
